### PR TITLE
Bean validation branch

### PR DESCRIPTION
**Pull request by irinaponomarevahaulmont**
Please review the original draft.

### CODE REVIEW

File: content/modules/studio/pages/bean-validation.adoc [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+The GraphQL schema in Amplicode-based projects contains directives corresponding to Bean Validation annotations. This functionality is present out of the box. As the schema is generated based on the backend objects, the fields of the input types get marked with Bean Validation annotations according to the rules set at the backend.
+
+All tools that interact with the GraphQL API can take these annotations into account. The annotations play the documenting role here.
+
+image::bean-validation/schema-update.png[align=center]
+
+[[frontend]]
+== Frontend: Bean Validation Support in React UI
+
+In React UI the *_Create_* and *_Edit_* screens contain built-in processing functionality for the Been Validation errors that come from the server.
```
Comment (alexbudarov) [inline content/modules/studio/pages/bean-validation.adoc:135]: Been  --> Bean

File: content/modules/studio/pages/bean-validation.adoc [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    @NotNull
+    @Column(name = "upc", nullable = false)
+    private Long upc;
+----
+
+
+[NOTE]
+If a DTO has been created for an entity, then all the customized messages should be duplicated in the DTO class; otherwise, they won't work. Therefore, we will copy the modified annotations for the `name` and `upc` fields to the DTO class.
+
+The dialog window for specifying the localized message will appear. Now we enter `Price must be positive` under *_Default_* and `Il prezzo deve essere positivo` under *_Italian_*.
```
Comment (alexbudarov) [inline content/modules/studio/pages/bean-validation.adoc:259]: Paragraphs order is messed a bit.
As I understand, this should follow directly after the "then click on the Intention (the bulb icon) and select Create or edit localized message"

File: content/modules/studio/pages/bean-validation.adoc [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+It matches the message that we configured for the `price` field via the DTO class.
+
+But if we try to edit the upc field and enter a value that consists of more than 12 digits, the message will be in English, because we have configured an unlocalized message for this particular error.
+
+image::bean-validation/unlocalized-in-italian-ui.png[align=center, width=700]
+
+The default messages with placeholders for parameters can be seen in the `ValidationMessages*.properties` files (*_Ctrl+Shift+N_* on Windows/Linux, *_⌘+Shift+O_* on macOS, or *_Navigate_* -> *_File_*).
+
+image::bean-validation/validation-messages.png[align=center]
```
Comment (alexbudarov) [inline content/modules/studio/pages/bean-validation.adoc:293]: screenshot is too big. Cut or scale.

File: content/modules/studio/pages/bean-validation.adoc [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+Inside these files, you can see the default messages for all languages supported by the Bean Validation library.
+
+*Example (Italian):*
+
+image::bean-validation/validation-messages-italian.png[align=center]
+
+This file is located inside the Hibernate Validator library, which can be found in the project's Tool Window.
+
+image::bean-validation/project-tool-window.png[align=center, width=450]
```
Comment (alexbudarov) [inline content/modules/studio/pages/bean-validation.adoc:303]: screenshot is too big. Cut or scale.

**Review by alexbudarov — CHANGES_REQUESTED** [review]

**Comment by irinaponomarevahaulmont** [comment]
Please review the changes

**Pull request merged**